test/integration/http_integration.cc

```diff
@@ -189,6 +191,7 @@ void HttpIntegrationTest::sendRequestAndWaitForResponse(Http::TestHeaderMapImpl&
                                                         uint32_t request_body_size,
                                                         Http::TestHeaderMapImpl& response_headers,
                                                         uint32_t response_size) {
+  response_.reset(new IntegrationStreamDecoder(*dispatcher_));
```

Comment:
Sure, this is a bit safer, I'll update my PR, as there are cleanups I'd also like, but I might as well do them.